apps/hash-api/src/graphql/resolvers/flows/runs.ts

```diff
@@ -0,0 +1,322 @@
+import type {
+  Client as TemporalClient,
+  WorkflowExecutionInfo,
+} from "@temporalio/client";
+import { temporal } from "@temporalio/proto";
+
+import type {
+  FlowRun,
+  FlowRunStatus,
+  QueryGetFlowRunsArgs,
+  ResolverFn,
+  StepRun,
+} from "../../api-types.gen";
+import { FlowStepStatus } from "../../api-types.gen";
+import type { GraphQLContext } from "../../context";
+
+const parseHistoryItemPayload = (
+  inputOrResults: temporal.api.common.v1.IPayloads | null | undefined,
+) =>
+  inputOrResults?.payloads?.map(({ data }) => {
+    if (!data?.toString()) {
+      return null;
+    }
+    return JSON.parse(data.toString());
+  });
```

Comment:
Just a thought, might it be better to filter out the `null` items in this array?

Reply:
Maybe, although I don't know in what circumstances we actually would see `null` here... let's leave it in for now and if we spot any we can deal with it because it might flag a problem.

Almost everything in the Temporal event types is possibly undefined, including things like ids which I think should be always defined.

Reply:
Oh I guess there's one in the last step in the example above... hmm. Maybe if `data` is undefined we filter it out, leave it if it's anything else falsy (including null)

Reply:
Filtered out undefined in https://github.com/hashintel/hash/pull/4218/commits/ff07002b94e48520f68cbffafacf24bf11eee9a6